File: Assets/ProjectGame/Lib/data/class_data/CharacterStats/BaseCharacterStats.py
from ..BaseCustomClassData import BaseCustomClassData

class BaseCharacterStats(BaseCustomClassData):
    def __init__(self):
        super().__init__()
        self.charisma = 0.0  # カリスマ
        self.intuition = 0.0  # 直感
        self.reasoning = 0.0  # ロジック
        self.appeal = 0.0  # 可愛さ
        self.deception = 0.0  # 演技力
        self.stealth = 0.0  # ステルス

    def read(self, reader):
        self.charisma = reader.read_float()
        self.intuition = reader.read_float()
        self.reasoning = reader.read_float()
        self.appeal = reader.read_float()
        self.deception = reader.read_float()
        self.stealth = reader.read_float()

    def load_json(self, data):
        self.charisma = data.get('charisma', 0.0)
        self.intuition = data.get('intuition', 0.0)
        self.reasoning = data.get('reasoning', 0.0)
        self.appeal = data.get('appeal', 0.0)
        self.deception = data.get('deception', 0.0)
        self.stealth = data.get('stealth', 0.0)

File: Assets/ProjectGame/Lib/data/class_data/CharacterStats/BaseCharacterStats.js
import { BaseCustomClassData } from '../BaseCustomClassData.js';

export class BaseCharacterStats extends BaseCustomClassData {
    constructor() {
        super();
        this.charisma = 0.0; // カリスマ
        this.intuition = 0.0; // 直感
        this.reasoning = 0.0; // ロジック
        this.appeal = 0.0; // 可愛さ
        this.deception = 0.0; // 演技力
        this.stealth = 0.0; // ステルス
    }

    read(view, offset) {
        let o = offset;
        this.charisma = view.getFloat32(o, true); o += 4;
        this.intuition = view.getFloat32(o, true); o += 4;
        this.reasoning = view.getFloat32(o, true); o += 4;
        this.appeal = view.getFloat32(o, true); o += 4;
        this.deception = view.getFloat32(o, true); o += 4;
        this.stealth = view.getFloat32(o, true); o += 4;
        return o;
    }

    loadJson(data) {
        this.charisma = data.charisma ?? 0.0;
        this.intuition = data.intuition ?? 0.0;
        this.reasoning = data.reasoning ?? 0.0;
        this.appeal = data.appeal ?? 0.0;
        this.deception = data.deception ?? 0.0;
        this.stealth = data.stealth ?? 0.0;
    }
}
